### ISSUE TITLE
Stack Overflow when parsing ajv-generated validation file

### ISSUE THREAD

**Issue by etinquis** [oven-sh/bun#7436]
### What version of Bun is running?

1.0.15+b3bdf22eb

### What platform is your computer?

Docker Desktop for Windows

### What steps can reproduce the bug?

https://github.com/etinquis/bun-0.15-repro

`docker --context desktop-linux run --rm -it --mount "type=bind,src=$PWD,dst=/home/bun/app" oven/bun:1.0.15 /bin/bash`

```
root@66276bb386d9:/home/bun/app# bun install
bun install v1.0.15 (b3bdf22e)
 + bun-types@1.0.15
 + @tsconfig/bun@1.0.1
 + ajv@8.12.0

 8 packages installed [3.65s]
root@66276bb386d9:/home/bun/app# bun run server.ts
Segmentation fault

```

### What is the expected behavior?

https://github.com/etinquis/bun-0.15-repro

`docker --context desktop-linux run --rm -it --mount "type=bind,src=$PWD,dst=/home/bun/app" oven/bun:1.0.14 /bin/bash`

```
root@f64e5f783172:/home/bun/app# bun install
bun install v1.0.14 (d8be3e51)
 + bun-types@1.0.15
 + @tsconfig/bun@1.0.1
 + ajv@8.12.0

 8 packages installed [3.44s]
root@f64e5f783172:/home/bun/app# bun run server.ts
starting server
```

### Additional information

model-validation.js is generated from json schemas via ajv https://ajv.js.org/standalone.html.

**Comment by etinquis** [comment]
Also fails on canary.

`docker --context desktop-linux run --rm -it --mount "type=bind,src=$PWD,dst=/home/bun/app" oven/bun:canary /bin/bash`

```
root@902adbcafae9:/home/bun/app# bun install
bun install v1.0.15 (03298cba)
 + bun-types@1.0.15
 + @tsconfig/bun@1.0.1
 + ajv@8.12.0

 8 packages installed [3.77s]
root@902adbcafae9:/home/bun/app# bun run server.ts
Segmentation fault
```

**Comment by Jarred-Sumner** [comment]
I'm pretty sure this is a stack overflow

Call stack:
```zig
src.js_parser.NewParser_(false,.react,false).parsePrefix
src.js_parser.NewParser_(false,.react,false).parseExprCommon
src.js_parser.NewParser_(false,.react,false).parseExpr
src.js_parser.NewParser_(false,.react,false).parseCallArgs
src.js_parser.NewParser_(false,.react,false).parseSuffix
src.js_parser.NewParser_(false,.react,false).parseExprCommon
src.js_parser.NewParser_(false,.react,false).parseExpr
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.NewParser_(false,.react,false).parseFnBody
src.js_parser.NewParser_(false,.react,false).parseFn
src.js_parser.NewParser_(false,.react,false).parseFnStmt
src.js_parser.NewParser_(false,.react,false).parseStmt
src.js_parser.NewParser_(false,.react,false).parseStmtsUpTo
src.js_parser.Parser._parse__anon_147687
src.js_parser.Parser.parse
src.cache.JavaScript.parse
src.bundler.Bundler.parseMaybeReturnFileOnlyAllowSharedBuffer__anon_151977
src.bun.js.module_loader.RuntimeTranspilerStore.TranspilerJob.run
src.bun.js.module_loader.RuntimeTranspilerStore.TranspilerJob.runFromWorkerThread
src.thread_pool.Thread.run
Thread.callFn__anon_160712
Thread.PosixThreadImpl.spawn__anon_109818.Instance.entryFn
```

**Comment by Jarred-Sumner** [comment]
looks like we need to increase the stack size for threads cc @dylan-conway

**Comment by Jarred-Sumner** [comment]
@etinquis as a temporary workaround, the main thread stack size is huge, so you should be able to change the `import` to `require` and that will transpile on the main thread and it should work

**Comment by Jarred-Sumner** [comment]
This was fixed by @dylan-conway in #7476
